packages/build-info/src/frameworks/remix.ts

```diff
@@ -1,24 +1,75 @@
-import { BaseFramework, Category, Framework } from './framework.js'
+import { BaseFramework, Category, DetectedFramework, Framework } from './framework.js'
+
+const CLASSIC_COMPILER_CONFIG_FILES = ['remix.config.js']
+const CLASSIC_COMPILER_DEV = {
+  command: 'remix watch',
+}
+const CLASSIC_COMPILER_BUILD = {
+  command: 'remix build',
+  directory: 'public',
+}
+
+const VITE_CONFIG_FILES = [
+  'vite.config.js',
+  'vite.config.mjs',
+  'vite.config.cjs',
+  'vite.config.ts',
+  'vite.config.mts',
+  'vite.config.cts',
+]
+const VITE_DEV = {
+  command: 'remix vite:dev',
+  port: 5173,
+}
+const VITE_BUILD = {
+  command: 'remix vite:build',
+  directory: 'build/client',
+}
 
 export class Remix extends BaseFramework implements Framework {
   readonly id = 'remix'
   name = 'Remix'
-  npmDependencies = ['remix', '@remix-run/netlify', '@remix-run/netlify-edge']
-  configFiles = ['remix.config.js']
+  npmDependencies = [
+    '@remix-run/react',
+    '@remix-run/dev',
+    '@remix-run/server-runtime',
+    '@netlify/remix-adapter',
+    '@netlify/remix-edge-adapter',
+    '@netlify/remix-runtime',
+    // Deprecated package name (deprecated in 1.6, removed in 2.0)
+    'remix',
+    // Deprecated Netlify packages
+    '@remix-run/netlify',
+    '@remix-run/netlify-edge',
+  ]
   category = Category.SSG
 
-  dev = {
-    command: 'remix watch',
-  }
-
-  build = {
-    command: 'remix build',
-    directory: 'public',
-  }
-
   logo = {
     default: '/logos/remix/default.svg',
     light: '/logos/remix/light.svg',
     dark: '/logos/remix/dark.svg',
   }
+
+  async detect(): Promise<DetectedFramework | undefined> {
+    await super.detect()
+
+    if (this.detected) {
+      const viteDetection = await this.detectConfigFile(VITE_CONFIG_FILES)
+      if (viteDetection) {
+        this.detected = viteDetection
+        this.dev = VITE_DEV
+        this.build = VITE_BUILD
+        return this as DetectedFramework
+      }
+      const classicCompilerDetection = await this.detectConfigFile(CLASSIC_COMPILER_CONFIG_FILES)
+      if (classicCompilerDetection) {
+        this.detected = classicCompilerDetection
+        this.dev = CLASSIC_COMPILER_DEV
+        this.build = CLASSIC_COMPILER_BUILD
+        return this as DetectedFramework
+      }
```

Comment:
This turned out not to be possible because when the "NPM dependency" detection gets "merged" with the "config file detection" it just picks the object with the highest accuracy and throws out the other, so we don't have `.config` on the detection object. I could have refactored to fix this, but... meh.